### PR TITLE
Add timing middleware

### PR DESCRIPTION
**Pull request by stuartcampbell**
Adding a simple timing middleware.  Using Starlette as this is more efficient (but less convenient) that the inbuilt FastAPI middleware.

### CODE REVIEW

File: src/nsls2api/middleware.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+        start_time = time.time()
+
+        async def send_wrapper(message: Message) -> None:
+            if message["type"] == "http.response.start":
+                headers = MutableHeaders(scope=message)
+                end_time = time.time()
```
Comment (danielballan) [inline src/nsls2api/middleware.py:21]:
```suggestion
                end_time = time.monotonic()
```

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.12
+# This file is autogenerated by pip-compile with Python 3.10
```
Comment (danielballan) [inline requirements-dev.txt:2]: Intentional step backward in Python version?

File: src/nsls2api/main.py [hunk 3]
```diff
@@ -56,17 +61,24 @@ def configure_routing():
...
+    app.include_router(home.router)
+    app.include_router(diagnostics.router)
+    app.mount("/static", StaticFiles(directory=static_root_absolute), name="static")
+    app.mount(
         "/assets",
         StaticFiles(directory=static_root_absolute / "assets"),
         name="assets",
     )
 
+# @api.middleware("http")
```
Comment (danielballan) [inline src/nsls2api/main.py:73]: May as well delete this, I suppose.

Good to know that the Starlette version of this is more efficient. Tiled stacks up a _bunch_ of these and probably could due to refactor them.

**Review by danielballan — COMMENTED** [review]
Minor suggestions

File: src/nsls2api/middleware.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    app: ASGIApp
+
+    def __init__(self, app: ASGIApp) -> None:
+        self.app = app
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+        start_time = time.time()
```
[the remark this answers is not in the document]
Comment (stuartcampbell) [reply src/nsls2api/middleware.py:16]: Or would time.perf_counter() be better still ?

File: src/nsls2api/middleware.py (hunk 1 above)
Comment (stuartcampbell) [reply src/nsls2api/middleware.py:21]: As this is for benchmarking, I've gone with `time.perf_counter()` to get the best accuracy.

**Review by stuartcampbell — COMMENTED** [review]
> Intentional step backward in Python version? 

Not intentional, just fighting with hatch to create the correct python virtual env for me! :)

**Pull request merged**